src/CoreBundle/private/library/classes/TCMSLogChange.class.php

```diff
@@ -1290,25 +1290,44 @@ public static function SetExtensionPosition($tableId, $sExtensionName, $sPreExte
     }
 
     /**
-     * moves a tab behind a given tab in cms_tbl_field_tab.
-     *
-     * @param int    $tableId
-     * @param string $sTabName
-     * @param string $sPreTabName - tabname where we want to set the new tab behind
+     * Moves a tab behind a given tab in cms_tbl_field_tab.
+     * Use the system names for the tabs.
+     * For backwards compatibility the name field is checked too.
      */
-    public static function SetTabPosition($tableId, $sTabName, $sPreTabName)
+    public static function SetTabPosition(string $tableId, string $tabSystemName, string $preTabSystemName): bool
     {
         // check position of field where we want to set the new field behind
-        $query = "SELECT * FROM `cms_tbl_field_tab` WHERE `name` = '".MySqlLegacySupport::getInstance()->real_escape_string($sPreTabName)."'";
-        $result = MySqlLegacySupport::getInstance()->query($query);
-        $posFieldRow = MySqlLegacySupport::getInstance()->fetch_assoc($result);
-        $pos = $posFieldRow['position'];
+        $databaseConnection = self::getDatabaseConnection();
+
+        $nameFieldWithTranslationSuffix = self::getFieldTranslationUtil()->getTranslatedFieldName('cms_tbl_field_tab', 'name');
+        
+        $query = "SELECT `position` FROM `cms_tbl_field_tab` WHERE `cms_tbl_conf_id` = :tableId AND (".$databaseConnection->quoteIdentifier($nameFieldWithTranslationSuffix)." = :preTabName OR `systemname` = :preTabName)";
+        $pos = $databaseConnection->fetchColumn($query, [
+                'tableId' => $tableId, 
+                'preTabName' => $preTabSystemName
+                ]
+        );
+        
+        if (false === $pos) {
+            self::addInfoMessage("Unable to position tab ".$tabSystemName." after ".$preTabSystemName." because ".$preTabSystemName." was not found", self::INFO_MESSAGE_LEVEL_ERROR);
+            
+            return false;
+        }
+
+        $pos = (int) $pos;
 
-        $query = "UPDATE `cms_tbl_field_tab` SET `position` = `position`+1 WHERE `position` > {$pos} AND `cms_tbl_conf_id` = '{$tableId}'";
+        $query = "UPDATE `cms_tbl_field_tab` SET `position` = `position`+1 WHERE `position` > ".$pos." AND `cms_tbl_conf_id` = ".$databaseConnection->quote($tableId);
```

Comment:
You can also write:

```
$query = "UPDATE `cms_tbl_field_tab` SET `position` = `position`+1 WHERE `position` > $pos AND `cms_tbl_conf_id` = ".$databaseConnection->quote($tableId);
```

which saves a couple of double quotes and dots in the middle.